storage/http_client.go

```diff
@@ -860,9 +860,21 @@ func (c *httpStorageClient) newRangeReaderXML(ctx context.Context, params *newRa
 			// Set custom headers passed in via the context. This is only required for XML;
 			// for gRPC & JSON this is handled in the GAPIC and Apiary layers respectively.
 			ctxHeaders := callctx.HeadersFromContext(ctx)
+
 			for k, vals := range ctxHeaders {
```

Comment:
Can we add a test for this logic? Could either use a mock transport, or factor out the header logic into a separate func for easy testing.